frontend/src/__mocks__/mockProjectK8sResource.ts

```diff
@@ -0,0 +1,39 @@
+import { ProjectKind } from '~/k8sTypes';
+
+type mockResourceConfigType = {
+  username: string;
+  displayName: string;
+  description: string;
+  k8sName: string;
+  enableModelMesh: boolean;
+};
+
+export const mockProjectK8sResource = (
+  config: mockResourceConfigType = {
+    username: 'test-user',
+    displayName: 'Test Project',
+    k8sName: 'test-project',
+    enableModelMesh: true,
+    description: '',
+  },
```

Comment:
🤔 The type is required but they have defaults? Not sure if TS is smart enough to run a partial config against the type... I suspect your defaults *never* get used.

Reply:
i got errors when i had them optional, i can give it another look